.github/workflows/typos.yml

```
@@ -0,0 +1,13 @@
+name: Check Typos
+on: [push, pull_request, workflow_dispatch]
```

```suggestion
  push:
    branches:
      - main
  pull_request:
  workflow_dispatch:
```
Currently, two runs run for each PR since it is triggered on pushes and the fact that this is PR